### ISSUE TITLE
Fix PipelineRun naming so trigger-pac-build targets the correct pipeline

### ISSUE THREAD

**Issue by jiridanek** [opendatahub-io/notebooks#3510]
## Context

The `trigger-pac-build` annotation works by having the build-service POST to PaC's `/incoming` endpoint with `pipelinerun: <component-name>-on-push`. PaC then searches `.tekton/` for a PipelineRun whose `metadata.name` matches that string.

Currently there are two mismatches:

**Main-branch push files** (`*-odh-main-push.yaml`):
- Component: `odh-pipeline-runtime-minimal-cpu-py312-ubi9` (branch=main)
- Build-service looks for: `odh-pipeline-runtime-minimal-cpu-py312-ubi9-on-push`
- Actual `metadata.name`: `odh-pipeline-runtime-minimal-cpu-py312-ubi9-odh-main-push`
- **Mismatch**: `-odh-main-push` vs `-on-push`

**Stable-branch push files** (`*-push.yaml`):
- Component: `odh-pipeline-runtime-minimal-cpu-py312-ubi9-ci` (branch=stable)
- Build-service looks for: `odh-pipeline-runtime-minimal-cpu-py312-ubi9-ci-on-push`
- Actual `metadata.name`: `odh-pipeline-runtime-minimal-cpu-py312-ubi9-on-push`
- **Mismatch**: component has `-ci` but pipelinerun name doesn't

**Base image additional complication**: some base image components have version-agnostic names (e.g. `odh-base-image-cuda-py312-c9s`) but the files have version-specific names (e.g. `odh-base-image-cuda-12-9-py312-c9s-odh-main-push`). Multiple files map to one component — these can't be fixed by renaming alone.

### 1. Rename stable push pipeline `metadata.name` fields (19 files)

Change from `<base-name>-on-push` to `<base-name>-ci-on-push` to match the `-ci` component name.

Example: in `odh-pipeline-runtime-minimal-cpu-py312-ubi9-push.yaml`:
```yaml
# before
name: odh-pipeline-runtime-minimal-cpu-py312-ubi9-on-push
# after
name: odh-pipeline-runtime-minimal-cpu-py312-ubi9-ci-on-push
```

**Files** (all `*-push.yaml` excluding `*-odh-main-push.yaml`):
- `odh-pipeline-runtime-{datascience-cpu,minimal-cpu,pytorch-cuda,pytorch-llmcompressor-cuda,pytorch-rocm,tensorflow-cuda,tensorflow-rocm}-py312-ubi9-push.yaml` (7)
- `odh-workbench-{codeserver-datascience-cpu,jupyter-datascience-cpu,jupyter-minimal-cpu,jupyter-minimal-cuda,jupyter-minimal-rocm,jupyter-pytorch-cuda,jupyter-pytorch-llmcompressor-cuda,jupyter-pytorch-rocm,jupyter-tensorflow-cuda,jupyter-tensorflow-rocm,jupyter-trustyai-cpu}-py312-ubi9-push.yaml` (11)
- No rstudio files (removed)

### 2. Rename main push pipeline `metadata.name` fields (non-base-image, ~18 files)

Change from `<base-name>-odh-main-push` to `<base-name>-on-push` to match the component name pattern build-service expects.

Example: in `odh-pipeline-runtime-minimal-cpu-py312-ubi9-odh-main-push.yaml`:
```yaml
# before
name: odh-pipeline-runtime-minimal-cpu-py312-ubi9-odh-main-push
# after  
name: odh-pipeline-runtime-minimal-cpu-py312-ubi9-on-push
```

### 3. Base image files — skip or defer

Base image components have version-agnostic names (`odh-base-image-cuda-py312-c9s`) but version-specific pipeline files (`cuda-12-9`, `cuda-13-0`). Multiple files map to one component, so `trigger-pac-build` can't know which to trigger. Skip these — they only build from main and are triggered by pathChanged() on real pushes.

## Verification

```bash
# Check that each component's expected pipelinerun name matches the metadata.name in .tekton/
# For stable:
for f in .tekton/*-push.yaml; do
  [[ "$f" == *-odh-main-push.yaml ]] && continue
  name=$(grep '^  name:' "$f" | head -1 | awk '{print $2}')
  comp=$(grep 'appstudio.openshift.io/component:' "$f" | head -1 | awk '{print $2}')
  expected="${comp}-on-push"
  [[ "$name" != "$expected" ]] && echo "MISMATCH: $f name=$name expected=$expected"
done

# For main (non-base-image):
for f in .tekton/*-odh-main-push.yaml; do
  [[ "$f" == *base-image* ]] && continue
  name=$(grep '^  name:' "$f" | head -1 | awk '{print $2}')
  comp=$(grep 'appstudio.openshift.io/component:' "$f" | head -1 | awk '{print $2}')
  expected="${comp}-on-push"
  [[ "$name" != "$expected" ]] && echo "MISMATCH: $f name=$name expected=$expected"
done
```

After renaming, `trigger-pac-build` should produce a matching PipelineRun for both main and stable components.

Also run the existing test suite:
```bash
./uv run pytest tests/test_main.py -x
```

**Comment by jiridanek** [comment]
@coderabbitai I encountered this issue while working on syncing the stable branch, and writing down docs 
* https://github.com/opendatahub-io/notebooks/pull/3509

**Comment by jiridanek** [comment]
@coderabbitai currently, annotating a `main` component can trigger `stable` branch build, the mismatches in naming have to be fixed

```
⏺ Bash(oc annotate components/odh-pipeline-runtime-minimal-cpu-py312-ubi9 build.appstudio.openshift.io/request=trigger-pac-build --context stone-rh01-odh -n open-data-…)
  ⎿  component.appstudio.redhat.com/odh-pipeline-runtime-minimal-cpu-py312-ubi9 annotated

⏺ Bash(sleep 30 && oc get pipelineruns --context stone-rh01-odh -n open-data-hub-tenant -l appstudio.openshift.io/component=odh-pipeline-runtime-minimal-cpu-py312-ubi9…)
  ⎿  NAME                                                              STATUS    STARTED
     odh-pipeline-runtime-minimal-cpu-py312-ubi9-odh-main-pull-hl6v8   Running   2026-05-05T05:53:58Z
```

⏺ Only the existing PR build. Let me wait a bit more and also check if the annotation was consumed:

```
⏺ Bash(oc get components/odh-pipeline-runtime-minimal-cpu-py312-ubi9 --context stone-rh01-odh -n open-data-hub-tenant -o jsonpath='{.metadata.annotations.build\.appstu…)
  ⎿  |{"pac":{"state":"enabled","configuration-time":"Fri, 18 Jul 2025 08:56:23 UTC"},"message":"done"}
```

⏺ Annotation consumed, PaC state enabled, but no PipelineRun. Same behavior as before — the build-service processes the annotation but the /incoming POST doesn't result in a PipelineRun.

```
⏺ Bash(sleep 60 && oc get pipelineruns --context stone-rh01-odh -n open-data-hub-tenant --sort-by=.metadata.creationTimestamp -o custom-columns=NAME:.metadata.name,COM…)
  ⎿  mod-arch-maas-on-pull-request-rmjq2                               mod-arch-maas                                         2026-05-05T06:11:51Z
     odh-pipeline-runtime-minimal-cpu-py312-ubi9-on-push-f7tfd         odh-pipeline-runtime-minimal-cpu-py312-ubi9-ci        2026-05-05T06:13:15Z
     odh-workload-variant-autoscaler-controller-on-push-rpmnj          odh-workload-variant-autoscaler-controller-ci         2026-05-05T06:13:19Z
     … +2 lines (ctrl+o to expand)  
```

⏺ Wait — look at that!                                                                                                                                                                                         

```
  odh-pipeline-runtime-minimal-cpu-py312-ubi9-on-push-f7tfd   odh-pipeline-runtime-minimal-cpu-py312-ubi9-ci   2026-05-05T06:13:15Z                                                                            
```

  A new PipelineRun appeared! It's for the -ci component (stable branch), triggered at 06:13. But we annotated the non--ci component (main). The build-service must have triggered the incoming webhook which  
  PaC matched against the stable push pipeline (since the pipelinerun name odh-pipeline-runtime-minimal-cpu-py312-ubi9-on-push matched the stable push file).                                                  
                                                                                                                                                                                                               
  Let me check its status:                                                                                                                                                                                     

```
⏺ Bash(oc get pipelinerun odh-pipeline-runtime-minimal-cpu-py312-ubi9-on-push-f7tfd --context stone-rh01-odh -n open-data-hub-tenant -o custom-columns=NAME:.metadata.n…)                                      
  ⎿  NAME                                                        STATUS    MESSAGE
     odh-pipeline-runtime-minimal-cpu-py312-ubi9-on-push-f7tfd   Running   Tasks Completed: 3 (Failed: 0, Cancelled 0), Incomplete: 19, Skipped: 0                                                             
```

⏺ It's running! trigger-pac-build DOES work — it just took about 2 minutes to kick in. The build-service needed time to ensure the incoming secret, update the Repository CR, and POST to the /incoming        
  endpoint.                                                                                                                                                                                                    
                                                                                                                                                                                                               
  So my earlier checks were just too impatient. The earlier attempt on the -ci component also likely worked but triggered the wrong pipeline (or we checked too early). And the rstudio one didn't work because
   rstudio was removed from main.                                                                                                                                                                              
                                                                                                                                                                                                               
  trigger-pac-build works — it just needs a couple of minutes to process. The annotation gets consumed quickly, but the actual PipelineRun creation happens asynchronously via the PaC incoming webhook.

**Comment by jiridanek** [comment]
@coderabbitai good findings, I investigated the 2 special cases (llmcompressor-cuda and tensorflow-rocm) in more depth

## Context

`trigger-pac-build` works by having the build-service POST to PaC's `/incoming` endpoint with `pipelinerun: <component-name>-on-push` and `branch: <component's configured branch>`. PaC searches `.tekton/` **on that branch** for a PipelineRun whose `metadata.name` matches. No collision risk between main/stable since PaC looks on the respective branch.

Currently all 36 non-base-image push pipeline files have mismatched names.

## The rule

`metadata.name` must equal `${component}-on-push`, where `component` is the registered Konflux component name (visible via `oc get components`).

## Changes

### 1. Fix stable push files (18 files on `main`, will reach `stable` via merge)

For each `*-push.yaml` (excluding `*-odh-main-push.yaml`): set `name: ${component}-on-push` using the component from the `appstudio.openshift.io/component` label.

One file also needs its **label fixed** — it doesn't match the registered component:
- `odh-pipeline-runtime-pytorch-llmcompressor-cuda-py312-ubi9-push.yaml`: label says `...-py312-ubi9-ci` but registered component is `odh-pipeline-runtime-pytorch-llmcompressor-cuda-py312-ci` (no `ubi9`). Fix the label, then derive the name from it.

Two files have component names that differ from the filename pattern (this is correct, matches what's registered):
- `*-pytorch-llmcompressor-cuda-py312-ubi9-push.yaml` → component `odh-workbench-jupyter-pytorch-llmcompressor-cuda-ci` → name `odh-workbench-jupyter-pytorch-llmcompressor-cuda-ci-on-push`
- `*-tensorflow-rocm-py312-ubi9-push.yaml` → component `odh-workbench-jupyter-tensorflow-rocm-py312-ci` → name `odh-workbench-jupyter-tensorflow-rocm-py312-ci-on-push`

### 2. Fix main push files (18 files, non-base-image)

For each `*-odh-main-push.yaml` (excluding `*base-image*`): change `name: <base>-odh-main-push` → `name: <base>-on-push`.

### 3. Base image files — skip

Base image components have version-agnostic names but version-specific pipelines. Multiple files per component. Can't fix with rename alone.

## Implementation

Script approach — read each file's component label, compute `${component}-on-push`, sed-replace the `name:` field:

```bash
# Stable push files
for f in .tekton/*-push.yaml; do
  [[ "$f" == *-odh-main-push.yaml ]] && continue
  comp=$(grep 'appstudio.openshift.io/component:' "$f" | head -1 | awk '{print $2}')
  expected="${comp}-on-push"
  old_name=$(grep '^  name:' "$f" | head -1 | awk '{print $2}')
  [[ "$old_name" != "$expected" ]] && sed -i "s|  name: ${old_name}|  name: ${expected}|" "$f"
done

# Main push files (non-base-image)
for f in .tekton/*-odh-main-push.yaml; do
  [[ "$f" == *base-image* ]] && continue
  comp=$(grep 'appstudio.openshift.io/component:' "$f" | head -1 | awk '{print $2}')
  expected="${comp}-on-push"
  old_name=$(grep '^  name:' "$f" | head -1 | awk '{print $2}')
  [[ "$old_name" != "$expected" ]] && sed -i "s|  name: ${old_name}|  name: ${expected}|" "$f"
done
```

**Before running**: fix the llmcompressor runtime label manually:
```
.tekton/odh-pipeline-runtime-pytorch-llmcompressor-cuda-py312-ubi9-push.yaml
  label: odh-pipeline-runtime-pytorch-llmcompressor-cuda-py312-ubi9-ci
  → fix to: odh-pipeline-runtime-pytorch-llmcompressor-cuda-py312-ci
```

## Verification

```bash
# 1. Check no mismatches remain (stable)
for f in .tekton/*-push.yaml; do
  [[ "$f" == *-odh-main-push.yaml ]] && continue
  name=$(grep '^  name:' "$f" | head -1 | awk '{print $2}')
  comp=$(grep 'appstudio.openshift.io/component:' "$f" | head -1 | awk '{print $2}')
  expected="${comp}-on-push"
  [[ "$name" != "$expected" ]] && echo "MISMATCH: $f"
done
# Should output nothing

# 2. Check no mismatches remain (main, non-base-image)
for f in .tekton/*-odh-main-push.yaml; do
  [[ "$f" == *base-image* ]] && continue
  name=$(grep '^  name:' "$f" | head -1 | awk '{print $2}')
  comp=$(grep 'appstudio.openshift.io/component:' "$f" | head -1 | awk '{print $2}')
  expected="${comp}-on-push"
  [[ "$name" != "$expected" ]] && echo "MISMATCH: $f"
done
# Should output nothing

# 3. Run tests
./uv run pytest tests/test_main.py -x
```
